src/file_system/mod.rs

```diff
@@ -354,6 +354,23 @@ impl Path {
     }
 }
 
+impl From<Vec<Label>> for Path {
+    fn from(labels: Vec<Label>) -> Self {
+        if labels.len() > 0 {
+            Self::from_labels(labels[0].clone(), &labels[1..])
+        } else {
+            panic!("Cannot convert empty vector to Path")
+        }
+    }
+}
+
+impl From<NonEmpty<Label>> for Path {
+    fn from(labels: NonEmpty<Label>) -> Self {
+        let (first, rest) = labels.split_first();
+        Self::from_labels(first.clone(), rest)
```

Comment:
This would actually be `Path(labels)`, since `Path` is just a newtype wrapper around `NonEmpty<Label>` :)